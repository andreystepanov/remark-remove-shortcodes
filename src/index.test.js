import removeShortcodes from './'
import unified from 'unified'
import rehype from 'rehype-parse'
import rehype2remark from 'rehype-remark'
import stringify from 'remark-stringify'

const parse = data =>
  unified()
    .use(rehype)
    .use(rehype2remark, {
      handlers: {
        comment: () => {},
      },
    })
    .use(removeShortcodes)
    .use(stringify, {
      emphasis: '*',
      strong: '*',
      bullet: '-',
      fences: true,
    })
    .processSync(data)
    .toString()

test('defined', () => {
  expect(removeShortcodes).toBeDefined()
  expect(typeof removeShortcodes).toBe('function')
})

// prettier-ignore
const data = `
  <main>
    <p>Some text</p>
    <p>[shortcode attr="1"]Shortcode content[/shortcode] and rest of the text...</p>
    <p>[shortcode attr="2"]Shortcode content 2[shortcode] and rest of the text...</p>
    <p>[shortcode_2 attr="false"/] Text after shortcode</p>
    <p>Text before shortcode [shortcode_3 attr="true"]</p>
    <p>Another content...</p>
  </main>
`

test('removes all WordPress shortcodes', () => {
  const md = parse(data)
  // console.log(md)
  expect(md).toMatchSnapshot()
})
