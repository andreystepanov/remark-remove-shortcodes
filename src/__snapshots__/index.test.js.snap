// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removes all WordPress shortcodes 1`] = `
"Some text

Shortcode content and rest of the text...

Shortcode content 2 and rest of the text...

Text after shortcode

Text before shortcode

Another content...
"
`;
